docs/source/spec-workload-executor.rst

```diff
@@ -43,29 +43,98 @@ After accepting the inputs, the workload executor:
    * MUST NOT override any of the URI options specified in the incoming connection string.
    * MUST NOT augment the incoming connection string with any additional URI options.
 
-#. MUST parse the incoming the ``driverWorkload`` document and use the ``MongoClient`` instance from the previous step
-   to run the operations described therein in accordance with the :ref:`test-scenario-format-specification`.
-   Note that the workload executor:
-
-   * MUST ignore the ``testData`` array. ``astrolabe`` is responsible for initializing the cluster with
+#. MUST parse the incoming the ``driverWorkload`` document and set up
+   the driver's unified test runner to execute the provided workload, with
+   the following deviations from the unified test runner specification:
+   
+   - Any `MongoClients <https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#entity-client>`_
+     that are instantiated by the workload executor MUST use the input
+     connection string as provided to the workload executor, and MUST
+     apply URI options specified in the particular test, if any, over the
+     provided connection string.
+   
+   - Each MongoClient MUST be set up to publish `command monitoring
+     <https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst>`_
+     events. The workload executor MUST record all events published
+     in the course of scenario execution.
```

Comment:
Looking at DriverWorkloadSubprocessRunner, `terminate()` issues `SIGINT` (or `CTRL_BREAK_EVENT` on Windows).

I expect most language are capable intercepting POSIX signals, but are we worried about Windows compatibility? For example, PHP's signal extension is not available on Windows and I'm not aware of any API to intercept `CTRL_BREAK_EVENT`.

Are all languages expected to run their workloads on Windows, or is Windows only supported as a possible platform for a subset of drivers?

Reply:
Ruby doesn't run on Windows at all hence I haven't run into this particular aspect of DAT.

Changing tte details of termination behavior I would say is outside the scope of this PR/spec, but I believe the following is true:

- As far as DAT is concerned, language is not required to run on a particular operating system. Any OS/es that the languages chooses to run on are fine.
- If additional changes are needed in astrolabe to support termination for new programming languages, they can in principle be made.